aws/table_aws_ses_template.go

```diff
@@ -0,0 +1,173 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ses"
+	"github.com/aws/aws-sdk-go-v2/service/ses/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableAwsSESTemplate(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ses_template",
+		Description: "AWS SES Template",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("name"),
+			Hydrate:    getSESTemplate,
+			Tags:       map[string]string{"service": "ses", "action": "GetTemplate"},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listSESTemplates,
+			Tags:    map[string]string{"service": "ses", "action": "ListTemplates"},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(AWS_EMAIL_SERVICE_ID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the template.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "subject_part",
+				Description: "The subject line of the email.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSESTemplate,
+			},
+			{
+				Name:        "text_part",
+				Description: "The email body that will be visible to recipients whose email clients do not display HTML.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSESTemplate,
+			},
+			{
+				Name:        "html_part",
+				Description: "The HTML body of the email.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSESTemplate,
+			},
+			{
+				Name:        "created_timestamp",
+				Description: "The time and date the template was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getSESTemplateARN,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSESTemplates(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := SESClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ses_template.listSESTemplates", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	maxItems := int32(1000)
+	// Limiting the results
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 1 {
+				maxItems = 1
+			} else {
+				maxItems = limit
+			}
+		}
```

Comment:
```suggestion
		if limit < maxItems {
			maxItems = limit
		}
```

I believe Steampipe handles cases where the `limit` value is less than `1`.